tests/test_transformations/test_knn_clustering.py

```diff
@@ -0,0 +1,205 @@
+import unittest
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+from sklearn.datasets import load_iris
+from spac.transformations import knn_clustering
+
+
+class TestKnnClustering(unittest.TestCase):
+    def setUp(self):
```

Comment:
@oeldaghar, can we start with a very simple case of minimum number of points (e.g., 4 points, two labeled and two unlabled, with K=1), then elaborate that use case to cover more corner cases?